slimt/Annotation.hh

```diff
@@ -249,6 +273,21 @@ class AnnotatedText {
   std::string_view as_view(const Range &range) const {
     return std::string_view(text.data() + range.begin, range.size());
   }
+  Encoding encoding_;
+};
+
+class WordIterator {
+ public:
+  explicit WordIterator(const AnnotatedText &annotated)
+      : annotated_(annotated) {}
+  Range operator*();
```

Comment:
Cant you do `Range&`, because this is pointing to `range_`?

Reply:
https://github.com/jerinphilip/slimt/pull/38/commits/763263be399ca4b676270896ebebe73993f25600